yield_plugin/include/yield_plugin/yield_plugin.hpp

```diff
@@ -234,14 +244,16 @@ class YieldPlugin
   void set_external_objects(const std::vector<carma_perception_msgs::msg::ExternalObject>& object_list);
 
   /**
-   * \brief Return collision time given two trajectories with one being predicted steps
+   * \brief Return naive collision time and locations based on collision radius given two trajectories with one being obstacle's predicted steps
+   * \param object_id object id corresponding to the trajectory2
    * \param trajectory1 trajectory of the ego vehicle
    * \param trajectory2 trajectory of predicted steps
    * \param collision_radius a distance to check between two trajectory points at a same timestamp that is considered a collision
    * NOTE: Currently Traj2 is assumed to be a simple cv model to save computational performance
-   * \return time_of_collision if collision detected, otherwise, std::nullopt
+   * NOTE: Collisions are based on only collision radius at the same predicted time even if ego vehicle maybe past the obstacle. To filter these cases, see `is_object_behind_the_vehicle()`
+   * \return data of time of collision if detected, otherwise, std::nullopt
    */
-  std::optional<rclcpp::Time> get_collision_time(const carma_planning_msgs::msg::TrajectoryPlan& trajectory1, const std::vector<carma_perception_msgs::msg::PredictedState>& trajectory2, double collision_radius);
+  std::optional<CollisionData> get_collision_data(uint32_t object_id, const carma_planning_msgs::msg::TrajectoryPlan& trajectory1, const std::vector<carma_perception_msgs::msg::PredictedState>& trajectory2, double collision_radius);
```

Comment:
1. It looks like this function no longer needs the `object_id` parameter.
2. Consider dropping the `_data` suffix. We are looking for a collision, not a collision data. The results, if there is a collision, are stored in a `GetCollisionData` object. This matches closer to the `std::mismatch` / `std::mismatch_result` naming style.

Reply:
1. Removed
2. replaced